python/pyarrow/tests/test_csv.py

```diff
@@ -974,6 +976,36 @@ def read_csv(self, *args, validate_full=True, **kwargs):
         table.validate(full=validate_full)
         return table
 
+    def test_row_numbers_in_errors(self):
+        """ Row numbers are only correctly counted in serial reads """
+        csv, _ = make_random_csv(4, 100, write_names=False)
```

Comment:
Is there a particular reason to only test without the header row? It'd be good to check that that gets handled properly too.